pkg/core/resources/apis/meshexternalservice/api/v1alpha1/helpers.go

```diff
@@ -59,3 +64,43 @@ func (t *MeshExternalServiceResource) Domains() *xds_types.VIPDomains {
 	}
 	return nil
 }
+
+func (t *MeshExternalServiceResource) GetPorts() []core.Port {
+	return []core.Port{
+		MesPort{
+			Value:    int(t.Spec.Match.Port),
+			Protocol: t.Spec.Match.Protocol,
+		},
+	}
+}
+
+func (t *MeshExternalServiceResource) FindPortByName(name string) (core.Port, bool) {
+	if name == strconv.Itoa(int(t.Spec.Match.Port)) {
+		return MesPort{
+			Value:    int(t.Spec.Match.Port),
+			Protocol: t.Spec.Match.Protocol,
+		}, true
+	}
+	return MesPort{}, false
+}
+
+type MesPort struct {
+	Value    int
+	Protocol core_mesh.Protocol
+}
+
+func (m MesPort) GetName() string {
+	return fmt.Sprintf("%d", m.Value)
+}
+
+func (m MesPort) GetValue() uint32 {
+	return uint32(m.Value)
+}
+
+func (m MesPort) GetNameOrStringifyPort() string {
+	return fmt.Sprintf("%d", m.Value)
```

Comment:
The implem here doesn't seem right no?

Also I think the API itself is broken because GetName should do this because that's the first thing folks will think about calling no?

Reply:
removed GetNameOrStringifyPort and left only GetName and documented it properly